Services/Devices.cs

```diff
@@ -174,17 +199,102 @@ public IDeviceClient GetClient(Device device, IoTHubProtocol protocol)
+
+            for (var batchNumber = 0; batchNumber < batches.Length; batchNumber++)
+            {
+                var batch = batches[batchNumber];
+
+                this.log.Info("Creating devices batch",
+                    () => new { batchNumber, batchSize = batch.Count() });
+
+                BulkRegistryOperationResult result = await this.registry.AddDevices2Async(
+                    batch.Select(id => new Azure.Devices.Device(id)), CancellationToken.None);
+
+                this.log.Info("Devices batch created",
+                    () => new { result.IsSuccessful, result.Errors });
+            }
+        }
+
+        /// <summary>
+        /// Delete a list of devices
+        /// </summary>
+        public async Task DeleteListAsync(IEnumerable<string> deviceIds)
+        {
+            this.SetupHub();
+
+            var batches = this.SplitArray(deviceIds.ToList(), REGISTRY_MAX_BATCH_SIZE).ToArray();
+
+            this.log.Info("Deleting devices",
+                () => new { Count = deviceIds.Count(), Batches = batches.Length, REGISTRY_MAX_BATCH_SIZE });
+
+            try
+            {
+                for (var batchNumber = 0; batchNumber < batches.Length; batchNumber++)
+                {
+                    var batch = batches[batchNumber];
+
+                    this.log.Info("Deleting devices batch",
+                        () => new { batchNumber, batchSize = batch.Count() });
+
+                    BulkRegistryOperationResult result = await this.registry.RemoveDevices2Async(
+                        batch.Select(id => new Azure.Devices.Device(id)),
+                        forceRemove: true,
+                        cancellationToken: CancellationToken.None);
+
+                    this.log.Info("Devices batch deleted",
+                        () => new { result.IsSuccessful, result.Errors });
+                }
+            }
+            catch (TooManyDevicesException error)
+            {
+                this.log.Error("Failed to delete devices, the batch is too big", () => new { error });
+                throw;
+            }
+            catch (IotHubCommunicationException error)
+            {
+                this.log.Error("Failed to delete devices (IotHubCommunicationException)", () => new { error.InnerException, error });
+                throw;
+            }
+            catch (Exception error)
+            {
+                this.log.Error("Failed to delete devices", () => new { error });
+                throw;
+            }
+        }
+
         /// <summary> 
         /// Get IoTHub connection string from either the user provided value or the configuration 
         /// </summary>
         public void SetCurrentIotHub()
         {
             string connString = this.connectionStringManager.GetIotHubConnectionString();
+
+            connString = "HostName=devissim003useast9dd22.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=ymxUjxVhTfqVLaT1SmZF7oETlWdTUyOR6cBE8npaGD0=";
```

Comment:
this looks like an actual key; I suspect you didn't mean to include this in your PR.